app/components/billing-process.js

```diff
@@ -0,0 +1,134 @@
+    },
+    set(key, value) {
+      return value;
+    }
+  }),
+  isStepOne: equal('currentStep', STEPS.stepOne),
+  isStepTwo: equal('currentStep', STEPS.stepTwo),
+
+  availablePlans,
+
+  showMonthly: not('showAnnual'),
+  defaultPlan: filterBy('availablePlans', 'isDefault'),
+  availablePlanNames: mapBy('availablePlans', 'name'),
+
+  monthlyPlans: computed('plans.@each.{name,annual,builds}', function () {
+    const { plans, availablePlanNames } = this;
+    const filteredMonthlyPlans = plans.filter(plan => {
+      const { annual, builds, name } = plan;
+      return !annual && builds <= 10 && availablePlanNames.includes(name);
+    });
+    const sortedMonthlyPlans = filteredMonthlyPlans.sort((a, b) => a.builds - b.builds);
+    return sortedMonthlyPlans;
+  }),
+
+  annualPlans: computed('plans.@each.{name,annual,builds}', function () {
+    const { plans, availablePlanNames } = this;
+    const filteredAnnualPlans = plans.filter(plan => {
+      const { annual, builds, name } = plan;
+      return annual && builds <= 10 && availablePlanNames.includes(name);
+    });
+    const sortedAnnualPlans = filteredAnnualPlans.sort((a, b) => a.builds - b.builds);
+    return sortedAnnualPlans;
+  }),
+
+  displayedPlans: computed(
+    'showAnnual',
+    'monthlyPlans.@each.{name,annual,builds}',
+    'annualPlans.@each.{name,annual,builds}',
+    function () {
+      const { annualPlans, showAnnual, monthlyPlans } = this;
+      return showAnnual ? annualPlans : monthlyPlans;
+    }),
+
+  selectedPlan: computed(
+    'displayedPlans.@each.{name,price,annual,builds}',
+    'defaultPlan', {
+      get() {
+        const { displayedPlans, defaultPlan } = this;
+        const plan = defaultPlan.get('firstObject');
+        return displayedPlans.findBy('name', plan.name);
+      },
+      set(key, value) {
+        return value;
+      }
+    }),
+
+  save: task(function* () {
+    try {
+      yield this.newSubscription.save();
+    } catch (error) {
+      this.flashes.error('There was an error saving the subscription task. Please try again.');
+    }
+  }).drop(),
+
+  actions: {
+
+    next() {
+      const { steps, currentStep } = this;
+      const index = steps.indexOf(currentStep);
+      this.set('currentStep', steps[index + 1]);
+    },
+
+    back() {
+      const { steps, currentStep } = this;
+      const index = steps.indexOf(currentStep);
+      this.set('currentStep', steps.get(index - 1));
+    },
+
+    cancel() {
+      this.reset();
+      this.set('currentStep', STEPS.stepOne);
+    },
+
+    handleSubmit(token, lastDigits) {
+      const { newSubscription, account } = this;
+      const organizationId = account.type === 'organization' ? account.id : null;
+      set(newSubscription, 'organizationId', organizationId);
+      set(newSubscription, 'plan', this.selectedPlan);
+      set(newSubscription.creditCardInfo, 'token', token);
+      set(newSubscription.creditCardInfo, 'lastDigits', lastDigits);
```

Comment:
We generally strive to stick with prototype version of `set` across the app. So, for consistency, I would recommend to use it as `newSubscription.creditCardInfo.set('lastDigits', lastDigits)`. And in case if you have to set multiple properties on the same object, there is a handy shorthand: 
```
newSubscription.creditCardInfo.setProperties({ token, lastDigits })
```